lib/adapters/http.js

```diff
@@ -3,16 +3,23 @@
 var utils = require('./../utils');
 var buildURL = require('./../helpers/buildURL');
 var transformData = require('./../helpers/transformData');
-var http = require('follow-redirects').http;
-var https = require('follow-redirects').https;
+var http = require('http');
+var https = require('https');
+var httpFollow = require('follow-redirects').http;
+var httpsFollow = require('follow-redirects').https;
 var url = require('url');
 var zlib = require('zlib');
 var pkg = require('./../../package.json');
 var Buffer = require('buffer').Buffer;
 
 // Resolve or reject the Promise based on the status
 function settle(resolve, reject, response) {
-  (response.status >= 200 && response.status < 300 ?
+  var minCode = 200;
+  var maxCode = 300;
+  if (response.config.hasOwnProperty('maxRedirects')) {
```

Comment:
I agree with @mzabriskie . On a side note, I'm going to submit a PR that adds support for a config option that allows specifying status codes for which an error should be thrown.